pkg/client.go

```diff
@@ -22,17 +24,38 @@ func NewApiClient(optionFuncs ...ClientOptionFunc) (*Client, error) {
 	rateLimitHandler := handlers.NewRateLimitHandler()
 	middlewares := options.Middleware
 	middlewares = append(middlewares, rateLimitHandler)
-	netHttpClient := kiotaHttp.GetDefaultClient(middlewares...)
+	defaultTransport := kiotaHttp.GetDefaultTransport()
+	netHttpClient := &http.Client{
+		Transport: defaultTransport,
+	}
+
 	if options.RequestTimeout != 0 {
 		netHttpClient.Timeout = options.RequestTimeout
 	}
 
+	// Configure GitHub App authentication if required fields are provided
+	if options.GitHubAppID != 0 && options.GitHubAppInstallationID != 0 && options.GitHubAppPemFilePath != "" {
+		existingTransport := netHttpClient.Transport
+		appTransport, err := ghinstallation.NewKeyFromFile(existingTransport, options.GitHubAppID, options.GitHubAppInstallationID, options.GitHubAppPemFilePath)
+		if err != nil {
+			return nil, fmt.Errorf("failed to create transport from GitHub App: %v", err)
+		}
+		netHttpClient.Transport = appTransport
```

Comment:
Per https://github.com/github/releases/issues/3898 you might want to accept the client ID here as well, preferably. 

It'd unfortunately be like 
```suggestion
	if (options.GitHubAppID != 0 || options.GitHubClientID) && options.GitHubAppInstallationID != 0 && options.GitHubAppPemFilePath != "" {
		existingTransport := netHttpClient.Transport
		appTransport, err := ghinstallation.NewKeyFromFile(existingTransport, options.GitHubAppID, options.GitHubAppInstallationID, options.GitHubAppPemFilePath)
		if err != nil {
			return nil, fmt.Errorf("failed to create transport from GitHub App: %v", err)
		}
		netHttpClient.Transport = appTransport
```
Etc...

Reply:
This is now done!